File: edupilot/lib/models/quiz/quiz.dart
import 'package:edupilot/models/quiz/difficulty.dart';
import 'package:edupilot/models/quiz/subject.dart';

class Quiz {

  // contructor
  Quiz({
    required this.id,
    required this.subject,
    required this.difficulty,
    required this.pointPerQuestion,
    required this.isActive,
  });

  // fields
  final String id;
  final Subject subject;
  final Difficulty difficulty;
  final int pointPerQuestion;
  final bool isActive;

}

File: edupilot_api/EduPilot.Api/Data/Models/Quiz.cs
namespace EduPilot.Api.Data.Models
{
    public class Quiz
    {
        public Guid Id { get; set; }
        public Guid PublisherId { get; set; }
        public List<Question> Questions { get; set; }
        public Guid SubjectId { get; set; }
        public Difficulty Difficulty { get; set; }
        public int Duration { get; set; }
        public int PointPerQuestion { get; set; }
        public bool IsActive { get; set; }
        public Subject Subject { get; set; }
        public Publisher Publisher { get; set; }
    }
}
